src/modules/Dimmer/Dimmer.js

```diff
@@ -23,165 +21,105 @@ const _meta = {
 
     /** An active dimmer will dim its parent container. */
-    active: customPropTypes.every([
-      PropTypes.bool,
-      customPropTypes.disallow(['disabled']),
-    ]),
-
-    /** A dimmable element can blur its contents. */
-    blurring: customPropTypes.every([
-      PropTypes.bool,
-      customPropTypes.demand(['dimmable']),
-    ]),
+    active: PropTypes.bool,
 
     /** Primary content. */
     children: PropTypes.node,
 
     /** Additional classes. */
     className: PropTypes.string,
 
-    closeOnOutsideClick: PropTypes.bool,
-
-    dimmable: customPropTypes.as,
-
-    dimmed: customPropTypes.every([
-      PropTypes.bool,
-      customPropTypes.demand(['dimmable']),
-    ]),
+    /** Shorthand for primary content. */
+    content: customPropTypes.contentShorthand,
 
-    /** A disabled dimmer cannot be activated. */
-    disabled: customPropTypes.every([
-      PropTypes.bool,
-      customPropTypes.disallow(['active']),
-    ]),
+    /** Handles click outside Dimmer's content, but inside Dimmer area. */
+    onClickOutside: PropTypes.func,
 
     /** A dimmer can be formatted to have its colors inverted. */
     inverted: PropTypes.bool,
 
-    /** Called when a close event happens */
-    onClose: PropTypes.func,
-
+    /** A dimmer can be formatted to be fixed to the page. */
     page: PropTypes.bool,
-  }
 
-  static defaultProps = {
-    closeOnOutsideClick: true,
+    /** A dimmer can be controlled with simple prop. */
+    simple: PropTypes.bool,
   }
 
   static _meta = _meta
 
-  handlePageMount = () => {
-    debug('handleMount()')
-
-    document.body.classList.add('dimmed', 'dimmable')
-  }
+  static Dimmable = DimmerDimmable
 
-  handlePageUnmount = () => {
-    debug('handleUnmount()')
+  handlePortalMount = () => document.body.classList.add('dimmed', 'dimmable')
 
-    document.body.classList.remove('dimmed', 'dimmable')
-  }
+  handlePortalUnmount = () => document.body.classList.remove('dimmed', 'dimmable')
 
-  handleOutsideClick = (e) => {
-    debug('handleOutsideClick()')
-    const { closeOnOutsideClick, onClose } = this.props
+  handleClickOutside = (e) => {
+    const { onClickOutside } = this.props
 
-    if (closeOnOutsideClick && onClose) onClose(e, this.props)
+    if (onClickOutside) onClickOutside(e, this.props)
   }
 
-  renderChildren() {
+  handleClickChildren = (e) => {
     const { children } = this.props
 
-    if (children) {
-      return (
-        <div className='content'>
-          <div className='center'>
-            {children}
-          </div>
-        </div>
-      )
-    }
-
-    return null
-  }
-
-  renderDimmable() {
-    const { blurring, children, className, dimmable, dimmed } = this.props
-
-    return (
-      <DimmerDimmable
-        blurring={blurring}
-        className={className}
-        Component={dimmable}
-        dimmed={dimmed}
-      >
-        {children}
-      </DimmerDimmable>
-    )
+    _.invoke(children, 'props.onClick', e)
+    e.stopPropagation()
```

Comment:
Do we need to stop propagation here?  Would prefer we not prevent handlers above this component from firing.

If we're trying to prevent our own handlers from firing, perhaps we should instead detect if the click was inside rather than stop the event.